src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -147,5 +151,11 @@ public class TagHelperDescriptor
         /// The <see cref="TagHelperUsageDescriptor"/> that indicates how to use the tag helper at design time.
         /// </summary>
         public TagHelperUsageDescriptor UsageDescriptor { get; }
+
+        /// <summary>
+        /// The HTML element the tag helper may output.
+        /// </summary>
+        /// <remarks>Overrides the HTML IntelliSense provided at design time.</remarks>
+        public string OutputElementHint { get; set; }
```

Comment:
What is the expected experience when multiple tag helpers target say `<joe>` and one's hint is "fred" but the other has no hint? How about both target '<joe>' and one's hint is "fred" but the other's is "ginger"?